sstables/consumer.hh

```diff
@@ -114,9 +119,10 @@ private:
     } _read_int;
 
     // state for READING_BYTES prestate
-    temporary_buffer<char> _read_bytes;
-    temporary_buffer<char>* _read_bytes_where; // which temporary_buffer to set, _key or _val?
-
+    size_t _read_bytes_len = 0;
+    std::vector<temporary_buffer<char>> _read_bytes;
```

Comment:
Can probably use small_vector here, possible follow up.

Reply:
I have another change to fix so I'm adding the small_vector as well in a new commit